sstables/sstable_set.cc

```diff
@@ -364,7 +364,12 @@ void partitioned_sstable_set::insert(shared_sstable sst) {
 }
 
 void partitioned_sstable_set::erase(shared_sstable sst) {
-    _all_runs[sst->run_identifier()].erase(sst);
+    if (auto it = _all_runs.find(sst->run_identifier()); it != _all_runs.end()) {
+        it->second.erase(sst);
+        if (it->second.empty()) {
+            _all_runs.erase(it);
+        }
+    }
```

Comment:
Should this be promoted into a separate pull request? It doesn't look related.

Reply:
It is.  See https://github.com/scylladb/scylladb/pull/14054
I dequeued this commit from the series in the next version.